### ISSUE TITLE
more information needed under: Couldn't retrieve the data for this visual. Please try again later.

### ISSUE THREAD

**Issue by fpk3** [MicrosoftDocs/powerbi-docs#4401]
**"Couldn't retrieve the data for this visual. Please try again later."**

This error is displayed consistently when a report consumer accesses a report which uses Direct Query over Power BI datasets connection and lacks build permission to the underlying dataset. 
https://learn.microsoft.com/en-us/power-bi/connect-data/desktop-directquery-datasets-azure-analysis-services#considerations-and-limitations
"Build permissions are required to view reports built using this feature when one or more datasets in the chain are in Pro workspaces." 

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: d6fe0913-78f9-47f1-ead1-dbecb17127d6
* Version Independent ID: 5828f2fc-25ae-18d8-a9ac-786511478fbe
* Content: [Troubleshooting tile errors - Power BI](https://learn.microsoft.com/en-us/power-bi/connect-data/refresh-troubleshooting-tile-errors)
* Content Source: [powerbi-docs/connect-data/refresh-troubleshooting-tile-errors.md](https://github.com/MicrosoftDocs/powerbi-docs/blob/main/powerbi-docs/connect-data/refresh-troubleshooting-tile-errors.md)
* Service: **powerbi**
* Sub-service: **pbi-data-sources**
* GitHub Login: @davidiseminger
* Microsoft Alias: **davidi**

**Comment by bishalgoswami** [comment]
Thanks for your comment, @fpk3. This issue has been assigned to the doc author for review.

**Comment by davidiseminger** [comment]
Thanks for your comment. This looks like a support request rather than documentation feedback. You might find a solution in the Power BI Community, found at https://community.powerbi.com. If not, you can get support by going to https://support.powerbi.com.

**Comment by fpk3** [comment]
Hi David,

This is NOT a support request, but I understand that you closed the issue.

I've seen this error many times, and it's *never* transient.
It always has to do with DirectQuery.

The most common cause of this error showing up was the requirement for
build permission for composite data models.
Now that composite models has gone GA, that requirement has been removed,
and therefore the error will not recur in that context.

These are the remaining two cases:
*#2. one dataset in composite model has been deleted.*
#3. Gateway data source offline for DirectQuery model.
For more information on the situations when this error occurs, please see:
https://www.fpk3.com/blog/power-bi-tile-error-couldnt-retrieve-data-visual-please-try-again-later

**Comment by jeroenterheerdt** [comment]
@fpk3 - the requirement for build permissions have _not_ been removed fully, so it looks like your Pro workspaces still require Build permissions. 
Quote from [the blog](https://powerbi.microsoft.com/en-us/blog/announcing-general-availability-for-composite-models-on-power-bi-datasets-and-analysis-services-models/):
"
When we started the preview, any consumer of a report that leveraged a composite model based on a Power BI dataset was required to have Build permissions or the Contributor role. Since then,[ we have changed this for Premium and PPU workspaces](https://powerbi.microsoft.com/blog/update-to-required-permissions-when-using-composite-models-on-a-power-bi-dataset/): readers of reports based on datasets in Premium or PPU workspaces just require Read permissions (or the Viewer role).

However, for most Pro workspaces Build permissions are still required in the scenario above. We are making backend changes to align the Pro workspaces with the Premium and PPU workspaces so everyone consuming these reports will just require Read permission or the Viewer role, regardless of the workspace type the data is stored in."
